### ISSUE TITLE
Some tags only contain one data point

### ISSUE THREAD

**Issue by Seraphli** [janosh/tensorboard-reducer#31]
`tb-reducer ant_sac/*seed*/log/serial/ -o out -r mean,std,min,max --lax-steps`
Although I used `--lax-steps` flag, some tags only contain one data point. Others are fine.
![image](https://user-images.githubusercontent.com/1586755/224345313-17616b4e-3ce0-4be9-9c55-0269f5439589.png)
![image](https://user-images.githubusercontent.com/1586755/224345411-7af48a0e-60b1-447b-b063-b7e7a3693976.png)

**Comment by janosh** [comment]
Can you share the log files so I can take a closer look?

**Comment by Seraphli** [comment]
Yes, you can download the logs from here.
https://transfer.sh/JDnbtA/ant_sac.tar.gz

**Comment by janosh** [comment]
Hmmm... Could you post your versions for Tensorboard Reducer, TensorBoard, and PyTorch/TensorFlow? Running the command you posted above

```sh
tb-reducer ant_sac/*seed*/log/serial/ -o out -r mean,std,min,max --lax-steps
```

on the data you shared, my TB dashboard looks very different.

Or maybe try reloading the dashboard a few times. I've known it to be a bit laggy in displaying all available data.

![Screenshot 2023-03-10 at 07 17 56](https://user-images.githubusercontent.com/30958850/224354376-9ab4ac79-2337-4cc2-b7c5-284ba4eb93aa.png)

**Comment by janosh** [comment]
Oops, my bad. Didn't read carefully enough. I'm not fully awake until I've had my 2nd tea or coffee. 😄 

I took another look and I think the problem is some of the runs are empty, resulting in those steps being filtered out.

If you re-run your command with [`--min-runs-per-step 1`](https://github.com/janosh/tensorboard-reducer#cli)

```sh
tb-reducer ant_sac/*seed*/log/serial/ -o out -r mean,std,min,max --lax-steps --min-runs-per-step 1 --overwrite
```


it does include many more steps in the reduction but it gives this jumbled mess which isn't very helpful either.

![Screenshot 2023-03-10 at 09 59 22](https://user-images.githubusercontent.com/30958850/224390528-f0c8c5f4-57cc-40af-b27b-19c312a355a3.png)

I'll have to dig deeper to see why that happens but maybe you can try removing/excluding the empty event files as workaround in the meantime.

**Comment by Seraphli** [comment]
Actually, all logs that match `ant_sac/*seed*/log/serial/` are not empty. I think again about this issue. It might be caused by the different logging time steps of the tag `collector_step/reward_mean`. The data points of the tag `evaluator_step/reward_mean` is always logged at the same time steps, while the data points of the tag `collector_step/reward_mean` is logged at random time steps. Is there a way to use some interpolations like linear interpolation or spline interpolation when reducing the logs?

**Comment by janosh** [comment]
Sorry for the long radio silence.

> The data points of the tag `evaluator_step/reward_mean` is always logged at the same time steps, while the data points of the tag `collector_step/reward_mean` is logged at random time steps.

Ah, that could be it. Glad you figured it out.

> Is there a way to use some interpolations like linear interpolation or spline interpolation when reducing the logs?

I haven't looked into that but happy to take a PR for this feature. You may be able to use the `pandas` [`interpolate` method](https://pandas.pydata.org/docs/reference/api/pandas.Series.interpolate.html) for this.
